src/MICore/Transports/LocalLinuxTransport.cs

```diff
@@ -80,7 +80,19 @@ public override void InitStreams(LaunchOptions options, out StreamReader reader,
                 throw new Exception(MICoreResources.Error_InvalidMiDebuggerPath);
             }
 
-            string debuggeeDir = System.IO.Path.GetDirectoryName(options.ExePath);
+            // Default working directory is next to the app
+            string debuggeeDir;
+            if (Path.IsPathRooted(options.ExePath) && File.Exists(options.ExePath))
+            {
+                debuggeeDir = System.IO.Path.GetDirectoryName(options.ExePath);
+            }
+            else
+            {
+                // If we don't know where the app is, default to HOME, and if we somehow can't get that, go with the root directory.
+                debuggeeDir = Environment.GetEnvironmentVariable("HOME");
+                if (string.IsNullOrEmpty(debuggeeDir))
+                    debuggeeDir = "/";
+            }
```

Comment:
Yeah, this code seems a bit weird to me too. That option setting will work because we issue a '-environment-cd' to GDB to set the working directory. This just establishes kind of an initial working directory. I think honestly this code should probably be removed and replaced with something (see debuggedprocess.cs(556)). But I didn't have a good way of proving out all the cases, so I was hesitant to just remove it. So I just wanted to avoid breaking what was here.